packages/blade/src/components/Box/BaseBox/useMemoizedStyles.web.ts

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import type { CSSObject } from 'styled-components';
+import { getBaseBoxStyles } from './getBaseBoxStyles';
+import type { BaseBoxProps } from './types';
+import type { Theme } from '~components/BladeProvider';
+
+const getDependencyProps = (props: BaseBoxProps & { theme?: Theme }): string | BaseBoxProps => {
+  // These are the props that change nothing in the getBaseBoxStyles calculations
+  const { theme, children, className, id, ...rest } = props;
+  let dependencyPropString: string | BaseBoxProps = '';
+  try {
+    dependencyPropString = JSON.stringify(rest);
```

Comment:
Some comment here about why the serialization would be nice.

Reply:
done